BraveWallet/Crypto/Accounts/AccountTransactionListView.swift

```diff
@@ -29,65 +28,56 @@ struct AccountTransactionListView: View {
-                       let url = baseURL?.appendingPathComponent("tx/\(tx.txHash)") {
-                      openWalletURL?(url)
-                    }
-                  }) {
-                    Label(Strings.Wallet.viewOnBlockExplorer, systemImage: "arrow.up.forward.square")
+    List {
+      Section(
+        header: WalletListHeaderView(title: Text(Strings.Wallet.transactionsTitle))
+      ) {
+        if activityStore.transactions.isEmpty {
+          emptyTextView(Strings.Wallet.noTransactions)
+        } else {
+          ForEach(activityStore.transactions) { tx in
+            Button(action: { self.transactionDetails = tx }) {
+              TransactionView(
+                info: tx,
+                keyringStore: keyringStore,
+                networkStore: networkStore,
+                visibleTokens: activityStore.assets.map(\.token),
+                allTokens: activityStore.allTokens,
+                displayAccountCreator: false,
+                assetRatios: assetRatios,
+                currencyCode: activityStore.currencyCode
+              )
+            }
+            .contextMenu {
+              if !tx.txHash.isEmpty {
+                Button(action: {
+                  if let baseURL = self.networkStore.selectedChain.blockExplorerUrls.first.map(URL.init(string:)),
+                     let url = baseURL?.appendingPathComponent("tx/\(tx.txHash)") {
+                    openWalletURL?(url)
                   }
+                }) {
+                  Label(Strings.Wallet.viewOnBlockExplorer, systemImage: "arrow.up.forward.square")
                 }
               }
             }
           }
         }
-        .listRowBackground(Color(.secondaryBraveGroupedBackground))
-      }
-      .navigationTitle(Strings.Wallet.transactionsTitle)
-      .navigationBarTitleDisplayMode(.inline)
-      .toolbar {
-        ToolbarItemGroup(placement: .confirmationAction) {
-          Button(action: { presentationMode.dismiss() }) {
-            Text(Strings.done)
-              .foregroundColor(Color(.braveOrange))
-          }
-        }
-      }
-      .sheet(item: $transactionDetails) { tx in
-        TransactionDetailsView(
-          info: tx,
-          networkStore: networkStore,
-          keyringStore: keyringStore,
-          visibleTokens: activityStore.assets.map(\.token),
-          allTokens: [],
-          assetRatios: assetRatios,
-          currencyCode: activityStore.currencyCode
-        )
       }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+    }
+    .listStyle(InsetGroupedListStyle())
+    .navigationTitle(Strings.Wallet.transactionsTitle)
+    .navigationBarTitleDisplayMode(.inline)
+    .sheet(item: $transactionDetails) { tx in
+      TransactionDetailsView(
+        info: tx,
+        networkStore: networkStore,
+        keyringStore: keyringStore,
+        visibleTokens: activityStore.assets.map(\.token),
+        allTokens: [],
```

Comment:
I believe we need to provide `activityStore.allTokens` here, or we won't display correct info for ERC20 approve transactions if the token approved is not a user visible token